File: Computer_Science/Common_Algorithms/Sorting/Selection_Sort/Selection_Sort.c
#include <stdio.h>

void selection_sort(int *arr, size_t size) {
  for (size_t i = 0; i < size; i++) {
    size_t smallest_idx = i;
    for (size_t j = i; j < size; j++) {
      if (arr[j] < arr[smallest_idx]) {
        smallest_idx = j;
      }
    }
    if (smallest_idx != i) {
      int temp = arr[smallest_idx];
      arr[smallest_idx] = arr[i];
      arr[i] = temp;
    }
  }
}

int main() {
  int arr[10] = {12, 15, 5, 1, 5, 18, 26, 4, 7, 13};
  size_t size = sizeof(arr) / sizeof(arr[0]);

  selection_sort(arr, size);

  for (int i = 0; i < size; i++) {
    printf("%d ", arr[i]);
  }
}

File: quantinium/Computer_Science/Common_Algorithms/Sorting/Selection_Sort/Selection_Sort.py
def selection_sort(arr):
    for i in range(0, len(arr)):
        smallest_idx = i
        for j in range(i, len(arr)):
            if arr[j] < arr[smallest_idx]:
                smallest_idx = j

        if smallest_idx != i:
            arr[i], arr[smallest_idx] = arr[smallest_idx], arr[i]
    return arr


def main():
    arr = [37, 12, 89, 5, 61, 23, 8, 45, 72, 16]
    selection_sort(arr)
    for num in arr:
        print(num)


main()
